client/ayon_maya/plugins/publish/extract_layout.py

```diff
@@ -99,13 +96,15 @@ def process(self, instance):
                 "instance_name": cmds.getAttr(
                     "{}.namespace".format(container)),
                 "representation": str(representation_id),
-                "version": str(version_id)
+                "version": str(version_id),
+                "extension": representation["name"]
```

Comment:
Note that the representation name does not have to be the extension. A representation may also have `representation["ext"]`, right @iLLiCiTiT ? Not sure if that is ALWAYS set though.

Reply:
I guess it's not supported yet?
![image](https://github.com/user-attachments/assets/3f6ad9bb-8cc9-412a-9132-0992f9a1f6a5)

Reply:
@moonyuet the error is not because it is not supported, but because you don't fill it. You can fill `"ext"` with extension and `"name"` with whatever you want.

NOTE: I don't know full context, let me know If I'm off.

Reply:
I just looked and you're working with existing representation entity. So the ext is in `representation["context"]["ext"]`.

NOTE: And also found out that it takes product type from representation context too, which is not really good idea. It should fetch product entity and use `productType` from it.